Userland/Libraries/LibPthread/pthread.cpp

```diff
@@ -685,47 +685,229 @@ int pthread_equal(pthread_t t1, pthread_t t2)
     return t1 == t2;
 }
 
+// FIXME: Use the fancy futex mechanism above to write an rw lock.
+//        For the time being, let's just use a less-than-good lock to get things working.
```

Comment:
cc @tomuta, @bugaevc - please don't kill me for the many millions of weird bugs here, it's my first time writing an rwlock, much less one that that conforms to the `pthread` APIs :P